### PR TITLE
Added rancheros-installer script

### PR DESCRIPTION
**Pull request by cloudnautique**
This script will call out to pull an os-installer container. It handles
orchestrating the install process.

The [os-installer repository](https://github.com/cloudnautique/os-installer) can be reviewed at the same time. There is an image on Docker hub already for the v0.1.1 release.

### CODE REVIEW

File: scripts/dockerimages/scripts/rancheros-install [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    exit 1
+fi
+
+if [[ ! -z "${CLOUD_CONFIG}" ]]; then
+    cp ${CLOUD_CONFIG} /opt/user_config.yml
+    CLOUD_INIT_DATA="/opt/user_config.yml"
+fi
+
+if [ "${FORCE_INSTALL}" != "true" ] && [ "${INSTALL_TYPE}" != "rancher-upgrade" ]; then
+    echo "This will wipe all data on ${DEVICE}!"
```
Comment (ibuildthecloud) [inline scripts/dockerimages/scripts/rancheros-install]: Do you want to print what you are about to do before you prompt?  Like "Partition: true\nDevice:/dev/sda"

File: scripts/dockerimages/scripts/rancheros-install [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      needed for SSH keys.
+  -d device
+  -f [ DANGEROUS! Data loss can happen ] partition/format without prompting
+  -p partition disk.
+        This will run fdisk and create one large ext4 partition.
+  -t install-type:
+        virtualbox-iso
+        amazon-ebs
+        rancher-upgrade
+  -v os-installer version.
```
Comment (ibuildthecloud) [inline scripts/dockerimages/scripts/rancheros-install]: is -v required?  What happens if I don't supply it.

File: README.md [hunk 3]
```diff
@@ -57,8 +57,51 @@ If you are running from the ISO RancherOS will be running from memory.  In order
...
+ ```
+ 
+ This command orchestrates installation from the rancher/os-installer. 
+ 
+####Examples:
+ Virtualbox installation:
+ 
+ `sudo rancheros-install -d /dev/sda -c ./cloud_data.yml -v v0.1.1 -t virtualbox-iso`
+ 
+ Upgrading:
```
Comment (ibuildthecloud) [inline README.md]: Lets not document upgrading at the moment

File: README.md [hunk 4]
```diff
@@ -57,8 +57,51 @@ If you are running from the ISO RancherOS will be running from memory.  In order
 
 ## Installing to Disk/Upgrading
 
-Coming soon (but you can guess it's all based on Docker)
+Beta:
 
+To install RancherOS on a new disk you can now use the `rancheros-install` command. 
+
+For non-ec2 installs, before getting started create a cloud-init file that will provide your initial ssh keys. At a minimum something like:
```
Comment (ibuildthecloud) [inline README.md]: Is the cloud config only used with the ec2 install?  I'll probably figure this out in a second when I read over the os-installer image...

File: README.md [hunk 5]
```diff
@@ -57,8 +57,51 @@ If you are running from the ISO RancherOS will be running from memory.  In order
 
 ## Installing to Disk/Upgrading
 
-Coming soon (but you can guess it's all based on Docker)
+Beta:
```
Comment (ibuildthecloud) [inline README.md]: Not needed, everything is beta :)

File: README.md [hunk 6]
```diff
@@ -57,8 +57,51 @@ If you are running from the ISO RancherOS will be running from memory.  In order
 
 ## Installing to Disk/Upgrading
```
Comment (ibuildthecloud) [inline README.md]: Want to change the title to just "Installing to Disk"

File: scripts/dockerimages/scripts/rancheros-install (hunk 2 above)
Comment (cloudnautique) [inline scripts/dockerimages/scripts/rancheros-install]: It is required, its used to pull in the container. For a default, I'm thinking latest.

File: README.md (hunk 3 above)
Comment (cloudnautique) [inline README.md]: Will remove upgrading stuff.

**Comment by cloudnautique** [comment]
@ibuildthecloud here is a version that just does the partitioning as part of the generic install. If we like this, then I can add a check that the device is the base block and not a partition.

File: README.md [hunk 7]
```diff
@@ -63,10 +63,44 @@ If you are running from the ISO RancherOS will be running from memory.  In order
...
+```
+Usage:
+  rancheros-install [options]
+Options:
+  -c cloud-config file
+      needed for SSH keys.
+  -d device
+  -f [ DANGEROUS! Data loss can happen ] partition/format without prompting
+  -t install-type:
+		generic
```
Comment (ibuildthecloud) [inline README.md]: remove tabs

**Pull request merged**